SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1447 @@
+	if ($doComparison -eq $false) {
+		$htmlOutput = $htmlOutput + "</p></H4><br><H4><p><font color=#FFA500>-Common Password Policy file path was not provided.  Drift data not available </font></p></H4>"
+	} else {
+		$htmlOutput = $htmlOutput + "</p></H4><br><p>Legend: drift data format <font color=#DC143C><i>Current Environment Value</i></font>[<i>standard Password Policy value</i>]</p>"
+	}
+	#prepare ESXi servers section"
+	$htmlOutput = $htmlOutput + '<H3>ESXi servers Password Policy</h3><table class="myTable">'
+	$tmpHtmlOutput = $ppmEnvironmentalDetails | where-Object { $_.productType -match "ESXi" } | select-object -Property FQDN -ExpandProperty productAttributes | select-object FQDN,passwdExpInDays,passwdMaxFailAttempts,passwdMinimumLengthFor1CharClass,passwdMinimumLengthFor2CharClass,passwdMinimumLengthFor3CharClass,passwdMinimumLengthFor4CharClass,passwdMinimumCharLengthForPhrase | sort-object -Property FQDN | ConvertTo-Html -Fragment -As Table
+	$htmlOutput += $tmpHtmlOutput
+	#prepare vCenter Servers section"
+	$htmlOutput = $htmlOutput + '<H3>vCenter Server Password Policy</h3><table class="myTable">'
+	$tmpHtmlOutput = $ppmEnvironmentalDetails | where-Object { $_.productType -match "VC" } | select-object -Property FQDN -ExpandProperty productAttributes | select-object FQDN,passwdExpInDays,passwdNotifyEmail | sort-object -Property FQDN | ConvertTo-Html -Fragment -As Table
+	$htmlOutput += $tmpHtmlOutput
+	#prepare SSO section"
+	$htmlOutput = $htmlOutput + '<H3>Single Sign On Password Policy</H3><p><font color=#6495ED>SSO FQDN: ' + $ppmVariables.ssoServerFqdn + '</font></p><table class="myTable">'
+	$tmpHtmlOutput = $ppmEnvironmentalDetails | where-Object { $_.productType -match "SSO" } | select-object -ExpandProperty productAttributes | select-object -property * -ExcludeProperty FQDN,productType,driftAlarm,driftMessage | ConvertTo-Html -Fragment -As Table
+	$htmlOutput += $tmpHtmlOutput
+	#prepare NSX Manager servers section"
+	$htmlOutput = $htmlOutput + '<H3>NSX Manager Password Policy</h3><table class="myTable">'
+	$tmpHtmlOutput = $ppmEnvironmentalDetails | where-Object { $_.productType -match "NSXMgr" } | select-object -Property FQDN -ExpandProperty productAttributes | select-object FQDN,passwdMinimumLength,apiPasswdMaxFailAttempts,apiPasswdMaxFailIntervalInSec,apiPasswdUnlockIntervalInSec,cliPasswdMaxFailAttempts,cliPasswdMaxFailIntervalInSec | sort-object -Property FQDN | ConvertTo-Html -Fragment -As Table
+	$htmlOutput += $tmpHtmlOutput
+	#prepare NSX Edge servers section"
+	$htmlOutput = $htmlOutput + '<H3>NSX Edge Password Policy</h3><table class="myTable">'
+	$tmpHtmlOutput = $ppmEnvironmentalDetails | where-Object { $_.productType -match "NSXEdge" } | select-object -Property FQDN -ExpandProperty productAttributes | select-object FQDN,passwdMinimumLength,cliPasswdMaxFailAttempts,cliPasswdMaxFailIntervalInSec | sort-object -Property FQDN | ConvertTo-Html -Fragment -As Table
```

Comment:
`Where-Object`

Reply:
addressed in bbfbaf6dac6bb885aa39e716c9d8971c0a5a58d7